sound/soc/sof/sof-acpi-dev.c

```diff
@@ -224,18 +224,12 @@ static int sof_acpi_probe(struct platform_device *pdev)
 	/* find machine */
 	mach = snd_soc_acpi_find_machine(desc->machines);
 	if (!mach) {
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_NOCODEC)
 		/* fallback to nocodec mode */
```

Comment:
why are you changing this? This has nothing to do with HDaudio?

Reply:
yes, but you need to add 
#ifndef CONFIG_SND_SOC_SOF_NOCODEC
return error
#endif  

I don't delete it, it should be selected for HDaudio, or pci_probe or acpic_probe will quit with error. And CONFIG_SND_SOC_SOF_NOCODEC has nothing to do with HDaudio, so it is by no means to keep it here.